XenModel/Alerts/PolicyAlert.cs

```diff
@@ -104,7 +104,7 @@ public PolicyAlert(long priority, string name, DateTime _time, string body, stri
         public PolicyAlert(IXenConnection connection, string body)
         {
             PolicyType = "VMPP";
-            numberOfVMsFailed = Text.Split('\n').Length;
+            numberOfVMsFailed = (Text == null) ? 0 : Text.Split('\n').Length;

```

Comment:
numberOfVMsFailed doesn't make sense for info or warn alerts, and it needs to be moved inside the block where we identified this alert as an error and only after the Text is assigned, in line 143 `Text = sb.ToString();`